plugin/adsk/scripts/mayaUsd_selectionUtils.py

```diff
@@ -0,0 +1,17 @@
+import ufe
+
+def expandPathToSelection(path):
+    """
+    Retrieves the path of the selection item that contains the provided path.
+    If there is no selection or no item in the selection is an ancestor of the item,
+    then the original path itself is returned.
+    """
+    selection = ufe.GlobalSelection.get()
+    ufePath = ufe.PathString.path(path)
+    if selection.containsAncestor(ufePath):
```

Comment:
Again, I would do this:
```
while not ufePath.empty():
    if selection.contains(ufePath):
        return ufe.PathString.string(ufePath)
    ufePath = ufePath.pop()
```

Reply:
This is not exactly what we need. But it's a good starting point: I need to actually go up until I'm *no longer* in the selection and keep the one just before not being in the selection. (Your suggestion would return to original sub-part.)

Reply:
But isn't it the case that if the original sub-part is in the selection, then we DO want to return it?

Reply:
Yes. Look at the new version of the code. I stop when the parent would not be in the selection, thus finding the highest in-selection node that contained the object.